addons/udes_stock_routing/models/stock_move.py

```diff
@@ -289,3 +289,75 @@ def _update_reserved_quantity(
             owner_id=owner_id,
             strict=strict,
         )
+
+    def _do_unreserve(self):
+        """
+        Restore original picking configuration of split picks.
+
+        When a split pick collection is unreserved, we want to put the split
+        pick moves back into the original "pick" picking, so that subsequent
+        reservation will execute the splitting process from the beginning.
+
+        If the original picking is not amendable we should put unreserved moves
+        into a new picking.
+        """
+        res = super()._do_unreserve()
+        Pick = self.env["stock.picking"]
+        Rule = self.env["stock.rule"]
+
+        res = super()._do_unreserve()
+        applicable_rules = Rule.search(
+            [
+                ("u_run_on_assign", "=", True),
+                ("picking_type_id", "in", self.picking_type_id.ids),
+            ]
+        )
+
+        # TODO What if no origin?
+        for origin, moves in self.groupby("origin"):
```

Comment:
This goupby should go into a method so it can be overriden.

Reply:
Done (fixed in a force-push, sorry)